packages/compiler/src/compiler/compile.test.ts

```diff
@@ -563,15 +381,41 @@ describe('conditional expressions', async () => {
         ${CUSTOM_TAG_START}/if${CUSTOM_TAG_END}
       </user>
     `
-    const result1 = await getCompiledText(prompt, {
-      foo: true,
+
+    const result1 = await render({
+      prompt: removeCommonIndent(prompt),
+      parameters: { foo: true },
     })
-    const result2 = await getCompiledText(prompt, {
-      foo: false,
+    const result2 = await render({
+      prompt: removeCommonIndent(prompt),
+      parameters: { foo: false },
     })
 
-    expect(result1).toBe('Foo!')
-    expect(result2).toBe('Bar!')
+    expect(result1.messages).toEqual([
+      {
+        role: MessageRole.user,
+        name: undefined,
+        content: [
+          {
+            type: 'text',
+            text: 'Foo!',
+          },
+        ],
+      },
+    ])
+    expect(result2.messages).toEqual([
+      {
+        role: MessageRole.user,
+        name: undefined,
+        content: [
+          {
+            type: 'text',
+            text: 'Bar!',
+          },
+        ],
+      },
+    ])
+    /* expect(result2).toBe('Bar!') */
```

Comment:
👀

Reply:
Is a left over. Check now we do an `toEqual` on `result2`